zerver/views/users.py

```diff
@@ -116,26 +116,30 @@ def update_user_backend(request, user_profile, email,
 # TODO: Since eventually we want to support using the same email with
 # different organizations, we'll eventually want this to be a
 # logged-in endpoint so that we can access the realm_id.
+@zulip_login_required
 def avatar(request, email_or_id, medium=False):
     # type: (HttpRequest, str, bool) -> HttpResponse
     """Accepts an email address or user ID and returns the avatar"""
+    is_email = False
     try:
         int(email_or_id)
     except ValueError:
-        get_user_func = get_user_profile_by_email  # type: Callable[..., UserProfile]
-    else:
-        get_user_func = get_user_profile_by_id
+        is_email = True
 
     try:
         # If there is a valid user account passed in, use its avatar
-        user_profile = get_user_func(email_or_id)
+        realm = request.user.realm
+        if is_email:
+            user_profile = get_user(email_or_id, realm)
```

Comment:
I think this needs to be another function that will fetch avatars correctly for system bot users. I think we may want a wrapper for `get_user` that allows anyone to look up entries in `get_cross_realm_emails`.